components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authcontext/JWTTokenGenerator.java

```diff
@@ -288,8 +288,8 @@ public void generateToken(OAuth2TokenValidationMessageContext messageContext) th
         }
 
         JWT jwt = null;
-        if(!JWSAlgorithm.NONE.equals(signatureAlgorithm)){
-            jwt = OAuth2Util.signJWT(claimsSet, signatureAlgorithm, tenantDomain);
+        if (!JWSAlgorithm.NONE.equals(signatureAlgorithm)) {
+            jwt = OAuth2Util.signJWT(claimsSet, signatureAlgorithm, tenantDomain, appDO.getOauthConsumerSecret());
```

Comment:
It is better to pass the appDO instead of the client secret since it is specific to signing with HMAC.

Reply:
If so there must be multiple implementations because each place utill is used app dao is not available in the context

Reply:
We anyway have to send the client secret after the above change. So in these multiple places, there should a way to get a client secret or we have to send null.

So either we can make the method accept a data object and send null if it is not available in the caller

OR

We can have an overloaded definition as you mentioned.

I'm +1 for an overloaded definition.